Dockerfile

```diff
@@ -13,29 +13,29 @@ RUN apt-get install --no-install-recommends -y build-essential && \
 
 # The python install is done in two steps to avoid re-installing all dependencies every
 # time the code changes
+# set the BUILD_CORE_SERVICE to non null to install additional service dependencies
+ARG BUILD_CORE_SERVICE
 COPY pyproject.toml poetry.lock README.rst CHANGES.rst Makefile /code/renku/
-WORKDIR /code/renku
-RUN poetry export --without-hashes -f requirements.txt --output /tmp/requirements.txt && \
-    pip install --no-cache-dir -r /tmp/requirements.txt && \
-    pip install --no-cache-dir poetry-dynamic-versioning==0.17.1
-
 COPY .git /code/renku/.git
 COPY renku /code/renku/renku
+WORKDIR /code/renku
+# hadolint ignore=SC2086
+RUN if [ -n "${BUILD_CORE_SERVICE}" ]; then export EXT_BUILD="-E service" ; fi && \
+    poetry export --without-hashes ${EXT_BUILD} -f requirements.txt --output /tmp/requirements.txt && \
+    pip install --no-cache-dir -r /tmp/requirements.txt
```

Comment:
Why export the `requirements.txt` when you can make poetry install everything in a virtualenv in the builder docker image stage and then just copy that virtualenv over in the final stage. That is what I do in the notebook service. It saves space on the final image because you can tell poetry to not install a bunch of uneeded things like setuptools, poetry itself and others.

See here for example: https://github.com/SwissDataScienceCenter/renku-notebooks/blob/master/git_services/Dockerfile.sidecar

You do not need poetry, pip or setuptools to run the core service.

You do not have to change this now if you dont want to. Just open an issue about it.

Reply:
I switched to using poetry, though I couldn't go with exactly the same approach as in the git_services sidecar.